app/javascript/app/components/table.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+function THeader(props) {
+  return (
+    <thead>
+      <tr>
+        {props.config.map((field, i) =>
+          <th key={'header-' + field.header}>
+            {field.header}
+          </th>
+        )}
+      </tr>
+    </thead>
+  );
+}
+
+function Row(props) {
+  return (
+    <tr key={'offer-' + props.offer_id + '-row'}>
+      {props.config.map((field, i) =>
+        <td key={'offer-' + props.offer_id + '-row-' + i}>
+          {field.data(props.offer)}
+        </td>
+      )}
+    </tr>
+  );
+}
+
+class AdminTable extends React.Component {
+  render() {
+    return (
+      <Table striped bordered condensed hover>
+        <THeader config={this.props.config} />
+        <tbody>
+          this.props.offers().map((offer) =>
```

Comment:
*applause* 
So what would be the best solution to this? Should  we just be working with immutable objects throughout the application, rather than trying to deal with JS Objects? 
Or should we take the approach you had in TAPP? Components work with JS Objects, but on the appState level everything is an immutable??